apps/backend/apps/admin/src/submission/submission.service.ts

```diff
@@ -312,4 +363,140 @@ export class SubmissionService {
       testcaseResult: results
     }
   }
+
+  async getAssignmentTitle(assignmentId: number): Promise<string | null> {
+    const assignment = await this.prisma.assignment.findUnique({
+      where: {
+        id: assignmentId
+      },
+      select: {
+        title: true
+      }
+    })
+    if (!assignment) {
+      throw new EntityNotExistException('Assignment')
+    }
+    return encodeURIComponent(assignment.title)
+  }
+
+  // async getZipFileSize(readStream: ReadStream): Promise<number> {
+  //   return new Promise((resolve, reject) => {
+  //     const chunks: Buffer[] = []
+
+  //     readStream.on('data', (chunk: Buffer) => {
+  //       chunks.push(chunk)
+  //     })
+
+  //     readStream.on('end', () => {
+  //       const fileSize = chunks.reduce((acc, chunk) => acc + chunk.length, 0)
+  //       resolve(fileSize)
+  //     })
+
+  //     readStream.on('error', () => {
+  //       reject(
+  //         new UnprocessableDataException(
+  //           'Error occurred during calculating file size.'
+  //         )
+  //       )
+  //     })
+  //   })
+  // }
+
+  async compressSourceCodes(
+    groupId: number,
+    assignmentId: number,
+    problemId: number
+  ) {
+    const assignmentProblemRecords =
+      await this.prisma.assignmentProblemRecord.findMany({
+        where: {
+          assignmentId,
+          problemId,
+          isSubmitted: true
+        },
+        select: {
+          userId: true
+        }
+      })
+    if (assignmentProblemRecords.length === 0) {
+      throw new EntityNotExistException('AssignmentProblem')
+    }
+
+    const submissionInfos = await Promise.all(
+      assignmentProblemRecords.map(async (record) => {
+        const submissionInfo = await this.getAssignmentLatestSubmissionInfo(
+          assignmentId,
+          record.userId,
+          problemId
+        )
+        return submissionInfo
+      })
+    )
+
+    const assignmentTitle = await this.getAssignmentTitle(assignmentId)
+    const dirPath = path.join(__dirname, assignmentTitle!)
+    mkdirSync(dirPath, { recursive: true })
+
+    submissionInfos.forEach((info) => {
+      const code = plainToInstance(Snippet, info.code)
+      const formattedCode = code.map((snippet) => snippet.text).join('\n')
+      const filename = `${info.user?.studentId}${LanguageExtension[info.language]}`
+      const filePath = path.join(dirPath, filename)
+      writeFile(filePath, formattedCode, (err) => {
+        if (err) {
+          console.error(err)
```

Comment:
로깅은 NestJS logger가 세팅되어있어서 그거 쓰는 걸 추천드립니다.

```suggestion
          this.logger.error(err, 'compressSourceCodes')
```

Logger 세팅은 email.service.ts 참고하면 좋을 거 같아요

```ts
@Injectable()
export class EmailService {
  private readonly logger = new Logger(EmailService.name)
```